src/effects/Echo.cpp

```diff
@@ -141,6 +141,31 @@ size_t EffectEcho::ProcessBlock(EffectSettings &,
    return blockLen;
 }
 
+
+
+struct EffectEcho::Validator
+   : DefaultEffectUIValidator
+{
+   Validator(EffectUIClientInterface& effect,
+      EffectSettingsAccess& access, EffectEchoSettings& settings)
+      : DefaultEffectUIValidator{ effect, access }
+      , mSettings{ settings }
+   {}
+   virtual ~Validator() = default;
+
+   Effect& GetEffect() const { return static_cast<Effect&>(mEffect); }
+
+   bool ValidateUI() override;
+   bool UpdateUI() override;
+
+   void PopulateOrExchange(ShuttleGui& S);
+
+   EffectEchoSettings mSettings;
```

Comment:
I reverse my previous comment at this line.

Do NOT put the validator into its final form at this commit.  Make mSettings a reference at this step.  Change it to no reference only at the last step.

Reply:
ok, done